aspnetcore/client-side/import-export-interop.md

```diff
@@ -0,0 +1,394 @@
+---
+title: Run .NET from JavaScript using `[JSImport]`/`[JSExport]` interop
+author: pavelsavara
+description: Learn how to run .NET from JavaScript using `[JSImport]`/`[JSExport]` interop.
+monikerRange: '>= aspnetcore-7.0'
+ms.author: riande
+ms.custom: mvc
+ms.date: 10/20/2022
+uid: client-side/import-export-interop
+---
+# Run .NET from JavaScript using `[JSImport]`/`[JSExport]` interop
+
+This article explains how to run .NET from JavaScript (JS) using `[JSImport]`/`[JSExport]` interop.
+
+Existing JS apps can use the expanded client-side WebAssembly support in .NET 7 to reuse .NET libraries from JS or to build novel .NET-based apps and frameworks.
+
+> [!NOTE]
+> This article focuses on running .NET from JS apps without any dependency on [Blazor](xref:blazor/index). 
+
+<!--
+
+    HOLD: ADD TO NOTE AFTER BLAZOR ARTICLE IS MERGED ...
+
+    For guidance on using `[JSImport]`/`[JSExport]` interop in Blazor WebAssembly apps, see <xref:blazor/js-interop/import-export-interop>.
+
+-->
+
+These approaches are appropriate when you only expect to run on WebAssembly (:::no-loc text="WASM":::). Libraries can make a runtime check to determine if the app is running on :::no-loc text="WASM"::: by calling <xref:System.OperatingSystem.IsBrowser%2A?displayProperty=nameWithType>.
+
+## Prerequisites
+
+[!INCLUDE[](~/includes/7.0-SDK.md)]
+
+## Namespace
+
+The JS interop API described in this article is controlled by attributes in the <xref:System.Runtime.InteropServices.JavaScript?displayProperty=fullName> namespace.
+
+## .NET SDK and WebAssembly workloads
+
+Install the latest version of the [.NET SDK](https://dotnet.microsoft.com/download/dotnet/).
+
+Install the `wasm-tools` workload, which brings in the related MSBuild targets. Note: You only need the `wasm-experimental` workload if you want to use the project templates.
+
+```dotnetcli
+dotnet workload install wasm-tools
+```
+
+Optionally, install the `wasm-experimental` workload, which contains experimental project templates for getting started with .NET on WebAssembly in a browser app (WebAssembly Browser App) or in a Node.js-based console app (WebAssembly Console App). This workload isn't required if you plan to integrate `[JSImport]`/`[JSExport]` interop into an existing JS app.
+
+```dotnetcli
+dotnet workload install wasm-experimental
+```
+
+For more information, see the [Experimental workload and project templates](#experimental-workload-and-project-templates) section.
+
+## App configuration
+
+To configure the app's project file (`.csproj`):
+
+* Target `net7.0` or later:
+
+  ```xml
+  <TargetFramework>net7.0</TargetFramework>
+  ```
+
+* Specify `browser-wasm` for the runtime identifier:
+
+  ```xml
+  <RuntimeIdentifier>browser-wasm</RuntimeIdentifier>
+  ```
+
+* Specify an executable output type:
+
+  ```xml
+  <OutputType>Exe</OutputType>
+  ```
+  
+  > [!NOTE]
+  > The framework may support library projects without an entry point in a future release of .NET.
+
```

Comment:
This note was just an FYI to you. I prefer to avoid promising future features when the future can always change.

```suggestion
```